packages/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx

```diff
@@ -0,0 +1,131 @@
+  public readonly id = "TreeUiitemsProvider";
+
+  private _imodel: IModelConnection;
+  private _activeView?: Viewport;
+  private _enablePreloading?: boolean;
+  private _enableElementsClassGrouping?: boolean;
+  private _allViewports?: boolean;
+  private _additionalTrees?: SelectableContentDefinition[];
+  private _modelsTreeProps?: {};
+  private _categoriesTreeProps?: {};
+  private _spatialTreeProps?: {};
+  private _modelsTreeReplacement?: () => React.ReactNode;
+  private _categoriesTreeReplacement?: () => React.ReactNode;
+  private _spatialTreeReplacement?: () => React.ReactNode;
+
+  constructor(props: TreeWidgetControlOptions) {
+    this._imodel = props.iModelConnection;
+    this._activeView = props.activeView;
+    this._enablePreloading = props.enablePreloading;
+    this._enableElementsClassGrouping = props.enableElementsClassGrouping;
+    this._allViewports = props.allViewports;
+    this._additionalTrees = props.additionalTrees;
+    this._modelsTreeProps = props.additionalProps?.modelsTree;
+    this._categoriesTreeProps = props.additionalProps?.categoriesTree;
+    this._spatialTreeProps = props.additionalProps?.spatialTree;
+    this._modelsTreeReplacement = props.treeReplacements?.modelsTree;
+    this._categoriesTreeReplacement = props.treeReplacements?.categoriesTree;
+    this._spatialTreeReplacement = props.treeReplacements?.spatialTree;
+  }
+
+  public provideWidgets(
+    _stageId: string,
+    stageUsage: string,
+    location: StagePanelLocation,
+    _section: StagePanelSection | undefined,
+  ): ReadonlyArray<AbstractWidgetProps> {
+    const widgets: AbstractWidgetProps[] = [];
+    if (
+      stageUsage === StageUsage.General &&
+      location === StagePanelLocation.Right
+    ) {
+
+      const modelsTreeComponent = (
+        <ModelsTreeComponent
+          iModel={this._imodel}
+          allViewports={this._allViewports}
+          activeView={this._activeView}
+          enablePreloading={this._enablePreloading}
+          enableElementsClassGrouping={this._enableElementsClassGrouping}
+          {...this._modelsTreeProps}
+        />
+      );
+
+      const categoriesTreeComponent = (
+        <CategoriesTreeComponent
+          iModel={this._imodel}
+          allViewports={this._allViewports}
+          activeView={this._activeView}
+          enablePreloading={this._enablePreloading}
+          {...this._categoriesTreeProps}
+        />
+      );
+
+      const spatialContainmentComponent = (
+        <SpatialContainmentTree
+          iModel={this._imodel}
+          enablePreloading={this._enablePreloading}
+          enableElementsClassGrouping={
+            this._enableElementsClassGrouping
+              ? ClassGroupingOption.Yes
+              : ClassGroupingOption.No
+          }
+          {...this._spatialTreeProps}
+        />
+      );
+
+      const trees: SelectableContentDefinition[] = [
+        {
+          label: TreeWidget.translate("visibilityWidget.modeltree"),
```

Comment:
These strings are missing.... 
Could you add them to https://github.com/imodeljs/viewer-components-react/blob/master/packages/tree-widget/public/locales/en/TreeWidget.json

Reply:
Please update https://github.com/imodeljs/viewer-components-react/blob/master/packages/tree-widget/src/components/TreeWidgetControl.tsx such that it doesn't use `UiFramework.translate`, but rather `TreeWidget.translate`